src/main/java/com/exasol/adapter/sql/SqlFunctionScalarExtract.java

```diff
@@ -6,17 +6,27 @@
 import java.util.List;
 
 public class SqlFunctionScalarExtract extends SqlNode {
+    private static final String EXTRACT = "EXTRACT";
     private String toExtract;
     private List<SqlNode> arguments;
 
     public SqlFunctionScalarExtract(String toExtract, List<SqlNode> arguments) {
-        assert(arguments.size() == 1);
+        if (arguments == null) {
```

Comment:
Code duplication with code I have seen above. Please generalize and extract to an `SqlArgumentValidator` class.

Reply:
done